skglm/solvers/bfgs.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+import scipy.optimize
+from numpy.linalg import norm
+
+from skglm.solvers import BaseSolver
+from skglm.datafits import BaseDatafit
+
+
+class BFGS(BaseSolver):
```

Comment:
LBFGS should perform better a priori, if you can change

Reply:
I wanted to conform with [`scipy` naming](https://docs.scipy.org/doc/scipy/reference/optimize.minimize-bfgs.html#optimize-minimize-bfgs) as well as the [reference, Chapter 6](https://link.springer.com/book/10.1007/978-0-387-40065-5).
Shall we rename it anyway?

Reply:
Need to check more carefully but I reckon scipy's BFGS uses BFGS, while I'm suggesting the limited memory variant, L-BFGS, that should be implemented here (in addition with box constraints that we ignore): https://docs.scipy.org/doc/scipy-0.18.1/reference/optimize.minimize-lbfgsb.html